File: apps/auto-wechat/backend/internal/domain/pipeline.go
package domain

import "time"

type RunStatus string

const (
	RunStatusQueued    RunStatus = "queued"
	RunStatusRunning   RunStatus = "running"
	RunStatusSucceeded RunStatus = "succeeded"
	RunStatusFailed    RunStatus = "failed"
)

type StepStatus string

const (
	StepStatusPending   StepStatus = "pending"
	StepStatusRunning   StepStatus = "running"
	StepStatusSucceeded StepStatus = "succeeded"
	StepStatusFailed    StepStatus = "failed"
	StepStatusSkipped   StepStatus = "skipped"
)

type PipelineStep string

const (
	StepCollect    PipelineStep = "collect"
	StepRank       PipelineStep = "rank"
	StepEnrich     PipelineStep = "enrich"
	StepEditor     PipelineStep = "editor"
	StepWriter     PipelineStep = "writer"
	StepIllustrate PipelineStep = "illustrate"
	StepLayout     PipelineStep = "layout"
	StepReview     PipelineStep = "review"
	StepCover      PipelineStep = "cover"
	StepPublish    PipelineStep = "publish"
)

var DefaultPipelineSteps = []PipelineStep{
	StepCollect,
	StepRank,
	StepEnrich,
	StepEditor,
	StepWriter,
	StepIllustrate,
	StepLayout,
	StepReview,
	StepCover,
	StepPublish,
}

type PipelineRun struct {
	ID               string
	Status           RunStatus
	PublishMode      string
	LayoutTemplateID *string
	ErrorMessage     *string
	DraftMediaID     *string
	PreviewHTML      *string
	StartedAt        *time.Time
	FinishedAt       *time.Time
	CreatedAt        time.Time
	UpdatedAt        time.Time
	Steps            []PipelineRunStep
}

type PipelineRunStep struct {
	Step         PipelineStep
	Status       StepStatus
	ErrorMessage *string
	StartedAt    *time.Time
	FinishedAt   *time.Time
}

type CreateRunInput struct {
	PublishMode      string
	TriggeredBy      string
	LayoutTemplateID *string
}

File: apps/auto-wechat/frontend/src/constants/pipeline.ts
export const RUN_STATUS = {
  QUEUED: 'queued',
  RUNNING: 'running',
  SUCCEEDED: 'succeeded',
  FAILED: 'failed',
} as const

export type RunStatus = (typeof RUN_STATUS)[keyof typeof RUN_STATUS]

export const STEP_STATUS = {
  PENDING: 'pending',
  RUNNING: 'running',
  SUCCEEDED: 'succeeded',
  FAILED: 'failed',
  SKIPPED: 'skipped',
} as const

export type StepStatus = (typeof STEP_STATUS)[keyof typeof STEP_STATUS]

export const PIPELINE_STEPS = [
  'collect',
  'rank',
  'enrich',
  'editor',
  'writer',
  'illustrate',
  'layout',
  'review',
  'cover',
  'publish',
] as const

export type PipelineStep = (typeof PIPELINE_STEPS)[number]

export const STEP_LABELS: Record<PipelineStep, string> = {
  collect: '采集',
  rank: '排序',
  enrich: '富化',
  editor: '选题',
  writer: '写作',
  illustrate: '配图',
  layout: '排版',
  review: '质检',
  cover: '封面',
  publish: '发布',
}

export const PUBLISH_MODE = {
  DRAFT_ONLY: 'draft_only',
} as const

export type PublishMode = (typeof PUBLISH_MODE)[keyof typeof PUBLISH_MODE]

export const POLL_INTERVAL_MS = 3000
